### DISCUSSION TITLE
Will netmaker support cilium as ingress/egress manager

### DISCUSSION THREAD

**Discussion by wibed** [gravitl/netmaker#1257] [General]
Is this on the todo list? or are there reasons not to

**Comment by afeiszli** [comment]
Not sure what you mean by this, can you expand? When we talk about ingress/egress in Netmaker, we're not talking about Kubernetes ingress/egress. Typically cilium is a layer that would live on top of a netmaker network, which should work.

**Reply by wibed** [reply]
so netmaker does not work inside kubernetes itself?

**Reply by afeiszli** [reply]
Netmaker works inside of Kubernetes, but it doesn't manipulate your existing networking stack inside of Kubernetes; it compliments it. So, you can have cilium as your service network layer, and then netmaker would provide ingress/egress to cilium. But it would not change the way cilium works.
